### ISSUE TITLE
What about transpilers and sourcemaps?

### ISSUE THREAD

**Issue by a-x-** [eslint/eslint#9725]
**feature request**

### Environment

* **ESLint Version:** 4.13.1
* **Node Version:** 8.9.1
* **npm Version:** 5.6.0

**parser**: Babel-ESLint

### Problem
Apply eslint error linting power (i'm not interesting in style checks here) to marginal languages with a pleasure (as fast as possible, convenient reports with correct `filename:line` references).

### Use-case
I have a large ancient coffeescript code base. And I have to modernise it gradually w/o any immediately rewriting.
I want to lint some common errors like undefined variables in coffeescript jsx (coffeescript@2 is compatible with jsx and es6) files.

### Current workaround solution
I set up weird workflow:
1. grab changed files
2. compile them with coffeescript@2
3. run eslint on the product

it's slow and report is hard to read (filename:line references are not expected)

### What the basic solution
Add optional sourcemaps support into eslint for reporting improving

**Update**: I looked for sourcemap problem and found an issue https://github.com/eslint/eslint/issues/45 and the [eslint-path-formatter](https://github.com/Bartvds/eslint-path-formatter)
But **it [does not work](https://github.com/Bartvds/eslint-path-formatter/issues/3)** (throws exceptions; updated 4yr ago) and **it's partial solution**: I don't want an another formatter, I want the default one or any else formatter after a sourcemap processing.

### The best solution as I see
Add transpilers support. What an .eslintrc.js might be:
```
module.exports = {
  "parser": "babel-eslint",
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    },
    "sourcemaps": true,
    "transpiler": "coffeescript-eslint",
    "transpilerOptions": {
        "inlineSourcemaps": true
    }
  },
  // ...
}
```

**Comment by a-x-** [comment]
I'll try create PR, but I need some advices and support.

Thank you for great tool! :)

**Comment by ilyavolodin** [comment]
I'm pretty sure this is not something we would want to add to ESLint. ESLint isn't supposed to be run on the compiled code, since once you use transpiler you have no control over the code, and can't fix errors that would be reported (in most cases).
What you are looking for, can be done by creating your own custom plugin with [processor](https://eslint.org/docs/developer-guide/working-with-plugins#processors-in-plugins). Preprocessor would compile your code and feed output to ESLint, postprocessor would use source map and ESLint errors to map them back to the original location.

**Comment by a-x-** [comment]
I'm agree, thank you!

I'm gonna try out this.

**Comment by j-f1** [comment]
@a-x- You might also be interested in [CoffeeLint](http://www.coffeelint.org/) which seems to lint the original CoffeeScript directly.

If you’re feeling particularly adventurous, you could [create a custom parser](https://eslint.org/docs/developer-guide/working-with-plugins#working-with-custom-parsers) that translates the CoffeeScript AST to an [ESTree](https://github.com/estree/estree)-compatible AST.

**Comment by a-x-** [comment]
I solved my problem with a kind of workaround. 

As I understand custom parser isn’t a my way (ast transpiling is much more hard task rather then using ready code transpiler)

I wrote plugin with preprocessor, but postprocessor is useless for me, because I cannot retrieve code content with an inline sourcemaps an use sourcemaps consumer for fixing paths and lines. 

So I fixed and forked ESLint-path-formatter. 

I think currently to pass sourcemaps from preprocessor to postprocessor using global cache hash-map.
